accepted/future-releases/constructor-tearoffs/feature-specification.md

```diff
@@ -402,6 +402,18 @@ A compile-time error occurs if a constructor tear-off denotes a generative const
 
 *A generative constructor declared in an abstract class may well be executed as part of the execution of a constructor tear-off, via a superinitializer. It cannot, however, be torn off on its own, because the execution of a constructor tear-off is an instance creation, and abstract classes do not have instances. There is no similar constraint on a factory constructor.*
 
+### Compile-time errors
+
+This section lists the compile-time errors associated with this feature. The _italic_ entries are commentary, which indicates that they are stated elsewhere already.
+
+It is compile-time error if an expression of the form `e.new <parameterPart>?` occurs, and `e` is not a type literal.
+
+_It is a compile-time error if an expression of the form `e <typeArguments>` occurs, and the static type of `e` is `Function` or `dynamic`._
```

Comment:
It's an error *unless*
- `e` is a type literal denoting a generic class, mixin or type alias declaration, or
- the static type of `e` is a generic function type.

Since neither `Function` nor `dynamic` satisfies that, they're included (but probably still worth calling out specifically).

Reply:
Those two requirements follow from the normal static analysis. I'm aiming at specifying the special cases which are otherwise given special treatment: We _do_ allow `f<int>()` if the static type of `f` is `Function` or `dynamic`, so why wouldn't we support that with a tearoff, that is, `f<int>`, as well?

Do